service/history/archivalQueueFactory.go

```diff
@@ -0,0 +1,140 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package history
+
+import (
+	"math"
+
+	"go.uber.org/fx"
+
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/service/history/archival"
+	"go.temporal.io/server/service/history/queues"
+	"go.temporal.io/server/service/history/shard"
+	"go.temporal.io/server/service/history/tasks"
+	"go.temporal.io/server/service/history/workflow"
+)
+
+const (
+	archivalQueuePersistenceMaxRPSRatio = 0.15
+)
+
+type (
+	archivalQueueFactoryParams struct {
+		fx.In
+
+		QueueFactoryBaseParams
+	}
+
+	archivalQueueFactory struct {
+		archivalQueueFactoryParams
+		QueueFactoryBase
+		archiver archival.Archiver
+	}
+)
+
+func NewArchivalQueueFactory(
+	params archivalQueueFactoryParams,
+) QueueFactory {
+	hostScheduler := queues.NewNamespacePriorityScheduler(
```

Comment:
I am not sure if we need resource isolation for archival queue and have one task channel for each ns. Probably `queues.NewPriorityScheduler` is enough and we just need to provide weight like 10 or 100 for that one channel.

Reply:
Now using `queues.NewPriorityScheduler` with this:

```go
ArchivalProcessorSchedulerRoundRobinWeights: dc.GetMapProperty(
  dynamicconfig.ArchivalProcessorSchedulerRoundRobinWeights,
  ConvertWeightsToDynamicConfigValue(DefaultActiveTaskPriorityWeight),
)
```

Please let me know if that looks right

Reply:
^ never mind, I'm just not setting the weights and it looks like they just use the default weights in that case which all other calls to `queues.NewPriorityScheduler` do